docs/contributing.md

```diff
@@ -355,6 +355,45 @@ We recommend at least these settings:
 extraArgs = ["+nightly"]
 ```
 
+### Alternative development setup with Nix
+
+If you have [Nix](https://nixos.org/) installed, you can use the provided
+`flake.nix` to get a development environment with all necessary tools
+pre-configured:
+
+```shell
+nix develop
+```
+
+This gives you a shell with:
+
+- A nightly Rust toolchain (including `rustfmt`, `rust-analyzer`, and
+  `rust-src`)
```

Comment:
rust-src isn't relevant imo. can probably just replace this whole section with a single sentence:

> Once inside the shell, you can run `cargo build`, `cargo test` etc. as usual. Other tools like `uv`, `cargo-insta`, `nextest` and more are installed, and so you should be able to follow any other recommendations in this guide as needed.